worldedit/code.lua

```diff
@@ -3,16 +3,22 @@
 
 --- Executes `code` as a Lua chunk in the global namespace.
 -- @return An error message if the code fails, or nil on success.
-function worldedit.lua(code)
-	local func, err = loadstring(code)
-	if not func then  -- Syntax error
-		return err
+function worldedit.lua(code, name)
+	local factory, err = loadstring("return function(p) " .. code .. " end")
+	if not factory then  -- Syntax error
+		return false, err
 	end
-	local good, err = pcall(func)
-	if not good then  -- Runtime error
-		return err
+	local func=factory()
+	local player=minetest.get_player_by_name(name)
+	local p={name=name, player=player}
+	if player then
+		p["pos"]=vector.round(player:get_pos())

```

Comment:
I would prefer the variables to be available directly

Reply:
fixed in 734d1a9

Reply:
I don't think there's any sense in adding even more variables, so that should be fine.